code/modules/mining/equipment/resonator.dm

```diff
@@ -114,3 +114,10 @@
 	. = ..()
 	transform = matrix()*1.5
 	animate(src, transform = matrix()*0.1, alpha = 50, time = 4)
+
+/obj/effect/resonance/proc/replicate(turf/M, creator, timetoburst) 
```

Comment:
you never actually call this anywhere?

Reply:
Is this called anywhere?

Also let's type M as `turf/closed/mineral` and have it `!istype(M)` return on the first line of this proc.